library/spdm_common_lib/context_data.c

```diff
@@ -104,16 +104,14 @@ return_status spdm_set_data(IN void *context, IN spdm_data_type_t data_type,
 			return RETURN_INVALID_PARAMETER;
 		}
 		if (parameter->location == SPDM_DATA_LOCATION_CONNECTION) {
-			spdm_context->connection_info.secured_message_version
-				.spdm_version_count = (uint8)(
-				data_size / sizeof(spdm_version_number_t));
-			copy_mem(spdm_context->connection_info
-					 .secured_message_version.spdm_version,
+			if ((uint8)(data_size /sizeof(spdm_version_number_t)) != 1) {

```

Comment:
Remove the cast to uint8, since if data_size is large the upper bits would be truncated.

Reply:
done